### ISSUE TITLE
validateToken doesn't work when authenticating with Auth0 

### ISSUE THREAD

**Issue by icaroharry** [auth0/webtask-tools#20]
I had this issue using the `Webtask.fromExpress(server).auth0()`: It wouldn't validate my token and I'd get an Unauthorized error.

It seems that it's because of the Base64 encoding, but I'm not sure. I think that the token does not come encoded from Auth0 and when the library tries to do it, the validation fails.

I solved it overwriting the `validateToken` function and removing the encoding, but I think that the library could check for that or have the option to not encode the token. Here's my solution:

```javascript
module.exports = Webtask.fromExpress(server).auth0({
  validateToken: function (ctx, req, token, cb) {
    let user;
    console.log(token)
    try {
      user = require('jsonwebtoken').verify(token, ctx.secrets.AUTH0_CLIENT_SECRET, {
        audience: ctx.secrets.AUTH0_CLIENT_ID,
        issuer: 'https://' + ctx.secrets.AUTH0_DOMAIN + '/'
      });
    } catch (e) {
      return cb({
        code: 401,
        message: 'Unauthorized: ' + e.message
      });
    }
    return cb(null, user);
  }
});
```

**Comment by steve-a-jones** [comment]
@icaroharry  Encountering this issue as well.

After reading the docs at https://webtask.io/docs/auth and viewing several examples showing usage of `wt.fromExpress(server).auth0()` with no configuration parameters I was under the assumption that it would just work out of the box if I passed the correct authorization header.

So I tried your approach (using `validateToken`) above and received the same error. And then I tried:
`cb(null, jwt.decode(token))` and it worked beautifully. My understanding is that this is not the correct approach though and that `jwt.verify` should be used  instead.

Playing around a bit more I also tried this inside `validateToken`:
`cb(null)` ... and it worked. 

Am I missing something here? I'm new to auth0 and webtask so maybe i'm misunderstanding the purpose of `validateToken`.

BTW, I am first obtaining the `idToken` from Auth0 Lock, which I subsequently pass to the webtask using the 'Authorization: Bearer xxxx' header.

Here are a few excerpts from the docs at https://webtask.io/docs/auth that let me to the assumptions above.

> Note that if you customize the createToken logic, you will also likely need to provide a custom logic to validate your token in validateToken described below.

I'm not doing this. ^

> By default, the JWT access token can be provided either as the Authorization: Bearer {accessToken} HTTP request header or the access_token={accessToken} URL query parameter.

I'm doing this ^

> If you don't provide your own implementation of validateToken, the default implementation will ensure the token is an id_token issued by Auth0 using the client secret configured when creating the webtask. All claims of the token will be used to construct the user profile.

I want this ^ .. but without defining the `validateToken` function it doesn't work.

Lastly,

I was led to research this problem in the first place because it was getting 401s from webtask containing this header:

`x-wt-error:Unauthorized: invalid algorithm`.

**Comment by karllhughes** [comment]
The built-in auth seems to work for me with HS256 encrypted tokens, but not RS256. This is not specified in the docs though, just something I've seen via trial and error.

**Comment by karllhughes** [comment]
Ok, I still don't know why this issue is happening as I'm not very experienced with the inner-workings of this library, but I was able to get the authentication to work with an RS256 algorithm by changing my `AUTH0_CLIENT_SECRET` to a Signing Certificate instead, for example:

```
AUTH0_CLIENT_SECRET="-----BEGIN CERTIFICATE-----\n<Signing Certificate Here>-----END CERTIFICATE-----"
```

Just make sure to use double quotes around the cert and add `\n` instead of line breaks.

**Comment by NotMyself** [comment]
Hi @icaroharry, @steve-a-jones, @karllhughes,

This issue has been solved with version [3.3.0](https://www.npmjs.com/package/webtask-tools). Sorry for the late reply.
